plugin/web/extensions/sidebar/popups/mobileDropdown/Catalog/index.tsx

```diff
@@ -95,9 +96,11 @@ const Catalog: React.FC<Props> = ({
   }, [backendURL, backendProjectName]);
 
   const handleOpenDetails = useCallback(
-    (data?: DataCatalogItem) => {
-      setSelectedDataset(data);
-      setPage("details");
+    (data?: DataCatalogItem | DataCatalogGroup) => {
+      if (data && "dataID" in data) {
```

Comment:
Doesn't this mean no folders will be selectable with this callback?

Reply:
@KaWaite , you are right, because on mobile we can't make folders selectable, it will open a full page, and make accessing files impossible.
So on mobile I made sure it keeps the old behavior.

Reply:
But in the mobile's catalog we aren't even including folders, right? So we shouldn't need to change anything in the mobile code as the folders won't be there to confuse things, right?

Reply:
@KaWaite , you are right, but we changed the `onSelect` code, which affects both mobile and desktop, so `onSelect` can now have `DataCatalogGroup ` as a parameter, so I've added this condition for mobile here so it won't affect this part.

Reply:
@basel-i Sure the onSelect can accept DataCatalogGroup, but this handleOpenDetails can and should only accept the DataCatalogItem. I think you would get a type error passing something like `dataset: Group | Item` in to something that only wants `Item`, but passing `dataset: Item` in to `dataset: Group | Item` shouldn't cause troubles. You know what I mean?